cocotb/handle.py

```diff
@@ -415,12 +421,14 @@ def __le__(self, value):
         return AssignmentResult(self, value)
 
     def __eq__(self, other):
-        if isinstance(other, SimHandleBase):
-            if self._handle == other._handle:
-                return 0
-            return 1
+        """Equality comparator for non-hierarchy objects
 
-        # Use the comparison method of the other object against our value
+        If ``other`` is not a :class:`SimHandleBase` instance the comparision
+        uses the comparison method of the ``other`` object against our
+        ``.value``.
+        """
+        if isinstance(other, SimHandleBase):
+            return SimHandleBase.__eq__(self, other)
         return self.value == other
 
     def __ne__(self, other):
```

Comment:
This needs fixing to handle `__eq__` returning `NotImplemented`

Reply:
Will mean that it automatically uses the `SimHandleBase.__eq__` operator?

Reply:
What if the other object is also a child of `SimHandleBase` and has replaced the `__eq__` operator?

Reply:
@eric-wieser could you have a look?

Reply:
Fixed